src/main/java/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,56 @@
+package basesyntax.supplier;
+
+import basesyntax.model.Circle;
+import basesyntax.model.Color;
+import basesyntax.model.Figure;
+import basesyntax.model.IsoscelesTrapezoid;
+import basesyntax.model.RectangleF;
+import basesyntax.model.RightTriangle;
+import basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_AMOUNT = 5;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final int MAX_RANDOM_NUMBER = 50;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public final Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_AMOUNT)) {
+            case 0:
+                return new Circle(getRandomInt(),
+                           colorSupplier.getRandomColor());
+            case 1:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                           getRandomDouble(),
+                           getRandomDouble(),
+                           getRandomDouble());
+            case 2:
+                return new RectangleF(colorSupplier.getRandomColor(),
+                           getRandomDouble(),
+                           getRandomDouble());
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                           getRandomDouble(),
+                           getRandomDouble());
+            case 4:
+                return new Square(colorSupplier.getRandomColor(),
+                           getRandomDouble());
+            default:
+                return getDefaultFigure();
```

Comment:
We can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability